src/ceph-volume/ceph_volume/util/disk.py

```diff
@@ -762,65 +676,63 @@ def is_locked_raw_device(disk_path):
     return 0
 
 
-def get_devices(_sys_block_path='/sys/block', _dev_path='/dev', _mapper_path='/dev/mapper'):
+def get_block_devs_lsblk():
+    '''
+    This returns a list of lists with 3 items per inner list.
+    KNAME - reflects the kernel device name , for example /dev/sda or /dev/dm-0
+    NAME - the device name, for example /dev/sda or
+           /dev/mapper/<vg_name>-<lv_name>
+    TYPE - the block device type: disk, partition, lvm and such
+
+    '''
+    cmd = ['lsblk', '-plno', 'KNAME,NAME,TYPE']
+    stdout, _stderr, _rc = process.call(cmd)
+    return [re.split(r'\s+', line) for line in stdout]
```

Comment:
What do you think about parsing json instead? Is this flag available on all platforms?

https://github.com/karelzak/util-linux/blob/beb61b07c20ab902fec883a4bd087c45d2742dea/misc-utils/lsblk.c#L1753

Reply:
My worry about relying on the json output is that the feature might not be available. Given that the output is quiet easy to parse anyway, I refrained from cross checking the lsblk versions on supported platforms.

Reply:
@sebastian-philipp 
I was wondering same last week and eventually found this note - https://github.com/ceph/ceph/blob/master/src/ceph-volume/ceph_volume/api/lvm.py#L21